### ISSUE TITLE
AAF adapter: animated parameters are omitted

### ISSUE THREAD

**Issue by timlehr** [PixarAnimationStudios/OpenTimelineIO#1135]
## Feature Request

Animated AAF properties are omitted during translation. We would want do add them back in.

## Description

**Updated: adressed issue of non-linear values**

Currently the AAF adapter does not transcribe any animated parameters / properties from the source AAF file. If a parameter is animated `aaf2.misc.VaryingValue` (e.g. animated grid effect), the parameter will be omitted from being translated to OTIO. This results in very inconsistent results based on the animation state of each property.

This issue is easily fixed, but it poses a big problem when it comes to backwards compatibility. How should animated values be stored in OTIO metadata without breaking the existing property translation? Ideally animated parameters would have been included in design efforts from the beginning, with static parameters having identical representation - even just with one sample. Making this change now, would potentially break a lot of things. 

Instead we propose a halfway solution, where animated properties get wrapped in a dictionary that contains samples and interpolation type. To address the issue with non-linear interpolation curves, we can utilize `pyaaf2` integrated baking functionality - baking one sample per frame.

```json
# current parameter translation, omitting all animated values

"Parameters": {
    "AFX_GRID_TYPE": 0,
    "AFX_SCALE_Y_U": 74
}

# proposed parameter translation, including animated values
# non-animated parameters stay the same for back-compatibility

"Parameters": {
    "AFX_GRID_TYPE": 0,
    "AFX_SCALE_Y_U": 74
    "AFX_SCALE_Z_U": {
        "_aaf_keyframed_property": true,
        "keyframe_baked_values": [
            [
                0.0,
                22.0
            ],
            [
                1.0,
                22.5
            ],
            ...
            [
                254.0,
                100.0
            ]
       ],
        "keyframe_interpolation": "Linear",
        "keyframe_values": [
            [
                0.0,
                22.0
            ],
            [
                40.0,
                100.0
            ]
       ]
    }
}

```

**Comment by ssteinbach** [comment]
I'm happy to discuss further in the TSC, but I feel like at least for now, this should remain in the domain of user-defined content, just for interchange within a pipeline and not part of the OTIO main project (yet):

- these kinds of values are embedded in a coordinate system. Is it a time curve?  Is it in the space of the clip?  of the track?  Of the output timeline?
- Given that they're in a coordinate system, we need tools for transforming other values into that coordinate system, and OTIO (at present) can't do that for any timelines that have time effects in them
- the evaluation engine for these kinds of curves is usually proprietary.  For a linear interpolation its probably fine but anything more complicated than that could be a problem if people want to take data from one app to another.

This is a great example of a broader problem we're working on around coordinate systems and coordinate system projections.  Can you provide examples files that we can reference with these kinds of parameters in them?  

We're going to open this problem space to the community soon, and this would be a good grounding example/use case for that.

**Comment by timlehr** [comment]
Thanks for the feedback @ssteinbach!

I addressed some of your concerns regarding non-linear interpolation curves by adding optional per-frame baking of the property value to the proposal. 

A working implementation is available as a draft #1151.
